ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala

```diff
@@ -98,13 +99,32 @@ class DBProcessRepository(val dbConfig: DbConfig, val modelVersion: ProcessingTy
       case CustomProcess(mainClass) => (None, Some(mainClass))
     }
 
-    def versionToInsert(latestProcessVersion: Option[ProcessVersionEntityData], processesVersionCount: Int, processingType: ProcessingType): Option[ProcessVersionEntityData] = latestProcessVersion match {
-      case Some(version) if version.json == maybeJson && version.mainClass == maybeMainClass => None
-      case _ => Option(ProcessVersionEntityData(id = processesVersionCount + 1, processId = processId.value,
-        json = maybeJson, mainClass = maybeMainClass, createDate = DateUtils.toTimestamp(now),
-        user = loggedUser.username, modelVersion = modelVersion.forType(processingType)))
+    //TODO: Move this normalization to DTO - GraphProcess
+    def normalizeJsonString(jsonString: String): Either[InvalidProcessJson, String] = parse(jsonString) match {
+      case Left(_) => Left(InvalidProcessJson(s"Invalid raw json string: $jsonString."))
+      case Right(json) => Right(json.noSpaces)
     }
 
+    def createProcessVersionEntityData(id: Int, processingType: ProcessingType, json: Option[String], maybeMainClass: Option[String]) = ProcessVersionEntityData(
+      id = id, processId = processId.value, json =json, mainClass = maybeMainClass, createDate = DateUtils.toTimestamp(now),
+      user = loggedUser.username, modelVersion = modelVersion.forType(processingType)
+    )
+
+    //In some reasons in lastVersion.json is string with whitespaces..
+    def isLastVersionContainsSameJson(lastVersion: ProcessVersionEntityData, maybeJson: Option[String]) =
+      lastVersion.json.map(normalizeJsonString) == maybeJson.map(normalizeJsonString)
+
+    def versionToInsert(latestProcessVersion: Option[ProcessVersionEntityData], processesVersionCount: Int, processingType: ProcessingType) =
+      (latestProcessVersion, maybeJson) match {
+        case (Some(version), _) if isLastVersionContainsSameJson(version, maybeJson) && version.mainClass == maybeMainClass =>
+          Right(None)
+        case (_, Some(json)) =>
+          normalizeJsonString(json)
```

Comment:
I think we don't need this, as we normalize in isLastVersionContainsSameJson anyway?